src/main/java/org/benetech/mathshare/controller/GlobalErrorHandling.java

```diff
@@ -0,0 +1,21 @@
+package org.benetech.mathshare.controller;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.context.request.WebRequest;
+
+@ControllerAdvice
+public class GlobalErrorHandling {
+
+    private final Logger logger = LoggerFactory.getLogger(GlobalErrorHandling.class);
+
+    @ExceptionHandler(IllegalArgumentException.class)
```

Comment:
Have you forgotten about HttpMessageNotReadableException or you omitted it purposely?

Reply:
who is throwing it and why should we catch it?

Reply:
You're right, it's currently not thrown, so we should omit it.